### PR TITLE
Remove support for GL_POINT_SMOOTH and GL_LINE_SMOOTH antialiasing

### PR DESCRIPTION
**Pull request by HybridDog**
These antialiasing techniques have been removed in OpenGL 3.1, they were often executed by the CPU, and Minetest does not use them. The OpenGL wiki recommends that we do not use this functionality in our program. https://www.khronos.org/opengl/wiki/Multisampling#Smooth_antialiasing

### CODE REVIEW

File: source/Irrlicht/COGLESDriver.cpp [hunk 1]
```diff
@@ -1798,33 +1796,16 @@ void COGLES1Driver::setBasicRenderStates(const SMaterial& material, const SMater
 	// Anti aliasing
 	if (resetAllRenderStates || lastmaterial.AntiAliasing != material.AntiAliasing)
 	{
-//		if (FeatureAvailable[IRR_ARB_multisample])
-		{
-			if (material.AntiAliasing & EAAM_ALPHA_TO_COVERAGE)
-				glEnable(GL_SAMPLE_ALPHA_TO_COVERAGE);
-			else if (lastmaterial.AntiAliasing & EAAM_ALPHA_TO_COVERAGE)
-				glDisable(GL_SAMPLE_ALPHA_TO_COVERAGE);
+//		if (FeatureAvailable[IRR_ARB_multisample]) {
```
Comment (sfan5) [inline source/Irrlicht/COGLESDriver.cpp:1799]:
```suggestion
```

**Comment by lhofhansl** [comment]
Tested. Works. Including FSAA. (Linux)

File: source/Irrlicht/COpenGLDriver.cpp [hunk 2]
```diff
@@ -2557,46 +2555,29 @@ void COpenGLDriver::setBasicRenderStates(const SMaterial& material, const SMater
 	}
 
 	// Anti aliasing
-	if (resetAllRenderStates || lastmaterial.AntiAliasing != material.AntiAliasing)
-	{
-		if (FeatureAvailable[IRR_ARB_multisample])
+	if (resetAllRenderStates
+			|| lastmaterial.AntiAliasing != material.AntiAliasing
+			&& FeatureAvailable[IRR_ARB_multisample]) {
```
Comment (sfan5) [inline source/Irrlicht/COpenGLDriver.cpp:2560]: on second look this is actually broken
needs to be `(A || B) && C`

Comment (HybridDog) [reply source/Irrlicht/COpenGLDriver.cpp:2560]: You're right; thanks.

**Pull request merged**